### PR TITLE
Fix validation

### PR DESCRIPTION
**Pull request by zyndiecate**
Fixes https://github.com/giantswarm/app-service/issues/609

deps 
- https://github.com/giantswarm/app-service/pull/618

### CODE REVIEW

File: v2_user_config.go [hunk 1]
```diff
@@ -36,12 +36,6 @@ func (ad *V2AppDefinition) UnmarshalJSON(data []byte) error {
 
 	result := V2AppDefinition(adc)
 
-	// validate app definition without validation context. validation context is
-	// given on server side to additionally validate specific definitions.
-	if err := result.Validate(nil); err != nil {
```
Comment (ewoutp) [inline v2_user_config.go]: We have to make sure that in all places where this is used, we add validation code.

- CLI 
- App-Service (in those calls that accept a definition)

**Comment by ewoutp** [comment]
Let's make issues to add the needed validation.
Then LGTM

**Comment by zyndiecate** [comment]
The only place we need to add validation is the CLI. See https://github.com/giantswarm/cli/pull/489. API just passes data through. App-Service already validates the service definition on creation. Something else missing?

**Comment by marians** [comment]
> The only place we need to add validation is the CLI.

Just to make sure: Is this validation of a user-provided service definition, like used in `swarm create`?

**Comment by zyndiecate** [comment]
Yes

**Comment by marians** [comment]
At some point in the near future we'll want to add service creation to the webclient. Then it would be convenient to have the validation login on the backend side, to avoid re-implementing this in JavaScript.

**Comment by zyndiecate** [comment]
> API just passes data through. App-Service already validates the service definition on creation.

**Comment by marians** [comment]
I have a guess what you want to tell me. Shutting up.

**Comment by ewoutp** [comment]
> API just passes data through 

This is currently not entirely correct since the API also parses the data (and therefore still calls the validate function until we use this user-config version).
There is already an issue to make the api really pass-through: https://github.com/giantswarm/api/issues/140

You're correct about the rest.

**Pull request merged**